src/SFA.DAS.RoatpAssessor/SFA.DAS.RoatpAssessor.Web/Services/AssessorOverviewOrchestrator.cs

```diff
@@ -55,6 +68,57 @@ public async Task<AssessorApplicationViewModel> GetOverviewViewModel(GetApplicat
             return viewmodel;
         }
 
+        public string SetSectionStatus(List<PageReviewOutcome> sectionPageReviewOutcomes)
+        {
+            var sectionStatus = string.Empty;
+            if(sectionPageReviewOutcomes != null && sectionPageReviewOutcomes.Any())
+            {
+                if (sectionPageReviewOutcomes.Count.Equals(1))
+                {
+                    sectionStatus = sectionPageReviewOutcomes[0].Status;
+                }
+                else
+                {
+                    var passStatusesCount = sectionPageReviewOutcomes.Where(p => p.Status == AssessorPageReviewStatus.Pass).Count();
+                    var failStatusesCount = sectionPageReviewOutcomes.Where(p => p.Status == AssessorPageReviewStatus.Fail).Count();
+                    var inProgressStatusesCount = sectionPageReviewOutcomes.Where(p => p.Status == AssessorPageReviewStatus.InProgress).Count();
+                    var noTagCount = sectionPageReviewOutcomes.Where(p => p.Status == null || p.Status == string.Empty).Count();
+                    var allPassOrFail = sectionPageReviewOutcomes.Count.Equals(passStatusesCount + failStatusesCount);
+
+                    if (sectionPageReviewOutcomes.Count.Equals(noTagCount)) // All empty
```

Comment:
take out all these elseifs and just return the value you're setting to sectionStatus as individual ifs.  Easier to read.

Reply:
When we are using ElseIf structure we can catch unhandled scenario, with multiple IFs we cannot.

Reply:
I agree in that it looks a bit hard to read and may need a better selection of variables, however the if elseif structure isn't too bad and has comments detailing what is done.

Perhaps raise this as part of the finessing ticket above